desktop/core/src/desktop/loglistener.py

```diff
@@ -139,28 +150,24 @@ def run(self):
       logging._releaseLock()
       server.serve_until_stopped()
 
-  return Server(ConfigSocketReceiver, ConfigStreamHandler, server_address, verify)
+    def stop(self):
+      udsstopListening()
+      self.ready.clear()
 
-def udsstopListening():
-  """
-  Stop the listening server which was created with a call to listen().
-  """
-  global _udslistener
-  logging._acquireLock()
-  try:
-    if _udslistener:
-      _udslistener.abort = 1
-      _udslistener = None
-  finally:
-    logging._releaseLock()
+    def stopped(self):
+      return self.ready.is_set()
+
+  return Server(ConfigSocketReceiver, ConfigStreamHandler, server_address, verify)
 
 def argprocessing(args=[], options={}):
-  parser = argparse.ArgumentParser(prog='runloglistener', description='What this program does', epilog='Text at the bottom of help')
-  parser.add_argument('-s', '--socket', dest='socket', action='store', default='/tmp/hue.uds')
+  parser = argparse.ArgumentParser(prog='loglistener', description='Run Log listener listening the unix domain socket.')
+  parser.add_argument('-s', '--socket', dest='socket', action='store', default='')
 
   opts = parser.parse_args()
-  if opts.socket:
+  if opts.socket != '':
```

Comment:
Change:
if opts.socket != '':
to:
if opts.socket:

Reply:
fixing it.